src/org/traccar/database/ComputedAttributesManager.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 Anton Tananaev (anton@traccar.org)
+ * Copyright 2017 Andrey Kunitsyn (andrey@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.database;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.traccar.Context;
+import org.traccar.helper.Log;
+import org.traccar.model.ComputedAttribute;
+
+public class ComputedAttributesManager {
+
+    private final DataManager dataManager;
+
+    private final Map<Long, Set<ComputedAttribute>> deviceComputedAttributes = new ConcurrentHashMap<>();
```

Comment:
I don't think that we should really cater for that use case. I think whatever is easier to implement. It can be random order, from device to group, or even throw exception if there are two attributes with the same key. I can't think of any real case when user would need multiple computed attributes on the same field.